src/App.js

```diff
@@ -1,79 +1,127 @@
 import React from 'react';
 
-import { getAll, getById } from './api/phone'
+import {getAll, getById} from './api/phone'
 import Basket from './Basket'
 import Filter from './Filter'
 import Catalog from './Catalog'
-
+import Viewer from './Viewer'
 import './App.css';
 
 
 class App extends React.Component {
-  constructor(props) {
-    super(props);
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            phones: getAll(),
+            selectedPhone: null,
+            basketItems: [],
+        };
+    }
+
+    addToBasket = (phoneId) => {
+        const items = this.state.basketItems;
+        const isItemPresent = items.filter(el => el.name === phoneId).length > 0;
```

Comment:
Let's find item's index instead.
```suggestion
        const itemIndex = items.findIndex((el) => el.name === phoneId);
```

Reply:
It will help you to avoid unneeded calculations below